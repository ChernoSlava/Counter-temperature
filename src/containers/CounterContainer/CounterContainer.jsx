import React, { useContext } from "react";
import { Counter } from "@components";
import { CounterContext } from '@contexts';

import { typeCss } from '@constants';

export const CounterContainer = () => {
    const {state} = useContext(CounterContext);
    let counter = state.counterValue;
    
    let type = typeCss.typeZero;
    
    if ( counter < 0 && counter > - 10) {
        type = typeCss.typeLessZero;
    } else if ( counter > 0 && counter < 10 ) {
        type = typeCss.typeAboveZero;
    } else if (counter >= 10 ) {
        type = typeCss.typeAboveTen;
    } else if (counter <= -10 ) {
        type = typeCss.typeLessTen;
    }

    return (
        <Counter title={counter} type={type}/>
    )
}