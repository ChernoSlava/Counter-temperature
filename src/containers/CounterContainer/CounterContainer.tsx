import React from 'react';
import { useSelector } from 'react-redux';

import { Counter, CounterType } from '@components';
import { getCounterContainerPropsSelector } from '@store';

export const CounterContainer: React.FC = (): JSX.Element => {
  const { value, type } = useSelector(getCounterContainerPropsSelector);

  return <Counter title={String(value)} type={type as CounterType} />;
};
