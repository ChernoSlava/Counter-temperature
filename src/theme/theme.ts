export const theme = {
  colors: {
    grey10: '#696867',

    blue10: 'rgb(19, 28, 192)',
    blue20: 'rgb(72, 133, 218)',
    blue30: 'rgb(98, 138, 196)',
    blue40: 'rgb(0, 69, 166)',

    red10: 'rgb(136, 29, 29)',
    red20: 'rgb(252, 38, 0)',
    red30: 'rgb(215, 104, 84)',

    black: 'black',
    white: 'white',
  },
};

export type ThemeType = typeof theme;
