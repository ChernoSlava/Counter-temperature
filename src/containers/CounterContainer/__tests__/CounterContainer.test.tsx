import { CounterContainer } from "../CounterContainer";
import { render } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { Store, AnyAction } from '@reduxjs/toolkit';

import { theme } from '../../../theme';
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import { InitialCounter } from "@store";

const mockStore = configureStore();

const getUI = (store: Store<any, AnyAction>): JSX.Element => {
  return (
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <CounterContainer />
      </Provider>
    </ThemeProvider>
  );
};

describe('CounterContainer test', () => {
  it('Should render zero', () => {
    const { container } = render(getUI(mockStore({
      ...InitialCounter,
    })))
    expect(container).toMatchSnapshot();
  });

  it('Should render lesszero', () => {
    const { container } = render(getUI(mockStore({
      ...InitialCounter,
      value: -5
    })))
    expect(container).toMatchSnapshot();
  });

  it('Should render lessten', () => {
    const { container } = render(getUI(mockStore({
      ...InitialCounter,
      value: -10
    })))
    expect(container).toMatchSnapshot();
  });

  it('Should render abovezero', () => {
    const { container } = render(getUI(mockStore({
      ...InitialCounter,
      value: 4
    })))
    expect(container).toMatchSnapshot();
  });

  it('Should render aboveten', () => {
    const { container } = render(getUI(mockStore({
      ...InitialCounter,
      value: 2038
    })))
    expect(container).toMatchSnapshot();
  });
})