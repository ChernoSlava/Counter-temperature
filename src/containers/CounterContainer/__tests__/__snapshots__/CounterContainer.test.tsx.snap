// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterContainer test Should render aboveten 1`] = `
.c0 {
  width: 200px;
  height: 200px;
  border-radius: 50%;
  border: 1px solid white;
  background-color: rgb(72,133,218);
  text-align: center;
  padding-top: 80px;
  font-size: 30px;
  color: white;
  background-color: rgb(252,38,0);
  -webkit-transition: linear 0.9s;
  transition: linear 0.9s;
}

<div>
  <div
    class="c0"
    type="aboveten"
  >
    2038
  </div>
</div>
`;

exports[`CounterContainer test Should render abovezero 1`] = `
.c0 {
  width: 200px;
  height: 200px;
  border-radius: 50%;
  border: 1px solid white;
  background-color: rgb(72,133,218);
  text-align: center;
  padding-top: 80px;
  font-size: 30px;
  color: white;
  background-color: rgb(215,104,84);
  -webkit-transition: linear 0.9s;
  transition: linear 0.9s;
}

<div>
  <div
    class="c0"
    type="abovezero"
  >
    4
  </div>
</div>
`;

exports[`CounterContainer test Should render lessten 1`] = `
.c0 {
  width: 200px;
  height: 200px;
  border-radius: 50%;
  border: 1px solid white;
  background-color: rgb(72,133,218);
  text-align: center;
  padding-top: 80px;
  font-size: 30px;
  color: white;
  background-color: rgb(0,69,166);
  -webkit-transition: linear 0.9s;
  transition: linear 0.9s;
}

<div>
  <div
    class="c0"
    type="lessten"
  >
    -10
  </div>
</div>
`;

exports[`CounterContainer test Should render lesszero 1`] = `
.c0 {
  width: 200px;
  height: 200px;
  border-radius: 50%;
  border: 1px solid white;
  background-color: rgb(72,133,218);
  text-align: center;
  padding-top: 80px;
  font-size: 30px;
  color: white;
  background-color: rgb(98,138,196);
  -webkit-transition: linear 0.9s;
  transition: linear 0.9s;
}

<div>
  <div
    class="c0"
    type="lesszero"
  >
    -5
  </div>
</div>
`;

exports[`CounterContainer test Should render zero 1`] = `
.c0 {
  width: 200px;
  height: 200px;
  border-radius: 50%;
  border: 1px solid white;
  background-color: rgb(72,133,218);
  text-align: center;
  padding-top: 80px;
  font-size: 30px;
  color: white;
  background-color: #696867;
  -webkit-transition: linear 0.9s;
  transition: linear 0.9s;
}

<div>
  <div
    class="c0"
    type="zero"
  >
    0
  </div>
</div>
`;
